### PR TITLE
implement kosmos daemonset

### PR DESCRIPTION
**Pull request by hanweisen**

### CODE REVIEW

**Comment by wuyingjun-lucky** [comment]
/assign
I will review it ASAP

**Comment by duanmengkk** [comment]
Do we need a meeting to discuss this PR？

**Comment by hanweisen** [comment]
> Do we need a meeting to discuss this PR？

Yes, we need. I'll prepare a meeting to demonstrate the function and code.

File: docs/proposals/globaldaemonset/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+### User Stories (Optional)
+
+## Design Details
+### Architecture
+![daemonset architecture](img/arch.png)
+####新增2个CRD   
+daemonset.kosmos.io: 用于描述跨集群daemonset的CRD  
+shadowdaemonset.kosmos.io: 为每一个集群创建一个shadowdaemonset，用于描述原生daemonset的CRD  
+####新增5个controller
+host_daemonset_controller: 功能与k8s中的daemonset_controller一样，只是调和对象由daemonset变为了主集群的hadowdaemonset.kosmos.io
```
Comment (duanmengkk) [inline docs/proposals/globaldaemonset/README.md]:
```suggestion
host_daemonset_controller: 功能与k8s中的daemonset_controller一样，只是调和对象由daemonset变为了主集群的shadowdaemonset.kosmos.io
```

File: pkg/clustertree/knode-manager/extensions/daemonset/host_daemon_controller.go [hunk 2]
```diff
@@ -0,0 +1,1465 @@
+// nolint:dupl
+package daemonset
+
+// The host_daemon_controller is developed based on the Kubernetes daemon_controller.go, mainly to change the reconciled objects from daemonsets to shadowdaemonset.kosmos.io.
+// For more detail on daemon_controller.go ref: https://github.com/kubernetes/kubernetes/blob/release-1.25/pkg/controller/daemon/daemon_controller.go
```
Comment (hanweisen) [inline pkg/clustertree/knode-manager/extensions/daemonset/host_daemon_controller.go:5]: add ref daemon_controller note @wuyingjun-lucky

**Pull request merged**